FirebaseAuthUI/FUIAuth.m

```diff
@@ -324,20 +338,32 @@ - (void)signInWithEmailHint:(NSString *)emailHint
             break;
           }
         }
-        [authProviderUI signOut];
-        [authProviderUI signInWithDefaultValue:emailHint
-                      presentingViewController:presentingViewController
-                                    completion:^(FIRAuthCredential *_Nullable credential,
-                                                 NSError *_Nullable error,
-                                                 FIRAuthResultCallback  _Nullable result) {
-          if (error) {
-            if (completion) {
-              completion(nil, error);
+
+        [FUIAuthBaseViewController showSignInAlertWithEmail:emailHint
+                                                   provider:authProviderUI
+                                   presentingViewController:presentingViewController
+                                              signinHandler:^{
+          [authProviderUI signOut];
+          [authProviderUI signInWithDefaultValue:emailHint
+                        presentingViewController:presentingViewController
+                                      completion:^(FIRAuthCredential *_Nullable credential,
+                                                   NSError *_Nullable error,
+                                                   FIRAuthResultCallback  _Nullable result) {
+            if (error) {
+              if (completion) {
+                completion(nil, error);
+              }
+              return;
             }
-            return;
+            [tempAuth signInAndRetrieveDataWithCredential:credential completion:completion];
+          }];
+        }
+                                              cancelHandler:^{
+          if (completion) {
+            completion(nil, originalError);
           }
-          [tempAuth signInAndRetrieveDataWithCredential:credential completion:completion];
-        }];
+         }
+        ];
```

Comment:
nit: these should probably be on the same line

Reply:
Done, thanks!